xlators/protocol/server/src/server-rpc-fops_v2.c

```diff
@@ -2914,6 +2920,17 @@ server4_setxattr_resume(call_frame_t *frame, xlator_t *bound_xl)
     if (state->resolve.op_ret != 0)
         goto err;
 
+    /* Let the namespace setting can happen only from special mounts, this
+       should prevent all mounts creating fake namespace. */
+    if ((frame->root->pid >= 0) &&
```

Comment:
Any reason to do this check after lookup?

Reply:
are you saying we would have done it before resolving the path?  Yes, could have been done. We can move it there.